great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -540,14 +619,29 @@ def _reconcile_rule_parameter_builder_configs(
                 parameter_builder_name
             ] = serialized_config
 
-        effective_parameter_builder_configs = nested_update(
-            effective_parameter_builder_configs,
-            {
+        if reconciliation_strategy == ReconciliationStrategy.NESTED_UPDATE:
+            effective_parameter_builder_configs = nested_update(
+                effective_parameter_builder_configs,
+                {
+                    parameter_builder_config[
+                        "expectation_type"
+                    ]: parameter_builder_config
+                    for parameter_builder_config in parameter_builder_configs
+                },
+                dedup=True,
+            )
```

Comment:
Nitpick - could we separate out the comprehension into it's own line and then take that result and plug it into the nested update?

Reply:
Please apply throughout.

Reply:
@cdkini -- DONE (in next commit). Thanks!